### PR TITLE
Remove hand-rolled contract glue code

### PR DESCRIPTION
**Pull request by spooktheducks**

### CODE REVIEW

**Review by RyanRHall — APPROVED** [review]

File: core/internal/mocks/flux_aggregator.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.5.1. DO NOT EDIT.
+// Code generated by mockery v2.4.0-beta. DO NOT EDIT.
```
Comment (RyanRHall) [inline core/internal/mocks/flux_aggregator.go:1]: you need to upgrade mockery

**Review by RyanRHall — CHANGES_REQUESTED** [review]

**Review by se3000 — COMMENTED** [review]
Looks good, but yeah need the abigen fixed.

File: core/internal/gethwrappers/generated/flags_wrapper/flags_wrapper.go [hunk 2]
```diff
@@ -1537,6 +1538,36 @@ func (_Flags *Flags) UnpackLog(out interface{}, event string, log types.Log) err
 	return _Flags.FlagsFilterer.contract.UnpackLog(out, event, log)
 }
 
+func (_Flags *Flags) ParseLog(log types.Log) (interface{}, error) {
+	abi, err := abi.JSON(strings.NewReader(FlagsABI))
```
Comment (RyanRHall) [inline core/internal/gethwrappers/generated/flags_wrapper/flags_wrapper.go:1542]: can we just export the abi from every contract as part of our wrapper extender? We do this `abi.JSON(strings.NewReader(contractABI)` plenty of times to warrant it.

Comment (spooktheducks) [reply core/internal/gethwrappers/generated/flags_wrapper/flags_wrapper.go:1542]: Sure, makes sense. I think I'll do it in a follow-up PR

File: core/services/fluxmonitor/flux_monitor.go [hunk 3]
```diff
@@ -451,7 +450,15 @@ func NewPollingDeviationChecker(
...
-	}, nil
+	}
+	// This is necessary due to the unfortunate fact that assigning `nil` to an
+	// interface variable causes `x == nil` checks to always return false. If we
+	// do this here, in the constructor, we can avoid using reflection when we
+	// check `p.flags == nil` later in the code.
+	if flags != nil && !reflect.ValueOf(flags).IsNil() {
+		pdc.flags = flags
+	}
+	return pdc, nil
```
Comment (RyanRHall) [inline core/services/fluxmonitor/flux_monitor.go:461]: nice, good improvement :+1:

**Review by RyanRHall — APPROVED** [review]
:shipit:

**Pull request merged**